### PR TITLE
Offline mode refactoring

### PR DESCRIPTION
**Pull request by eroshenkoam**

### CODE REVIEW

**Comment by eroshenkoam** [comment]
test this please

File: src/main/java/ru/yandex/qatools/allure/repository/XmlConverter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * eroshenkoam
+ * 31.07.15
+ */
+public class XmlConverter implements Converter {
+
+    @Override
+    public Object fromBody(TypedInput typedInput, Type type) throws ConversionException {
+        Object result;
+
+        try {
```
Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/repository/XmlConverter.java]: use try-with-resources

File: src/main/java/ru/yandex/qatools/allure/repository/XmlConverter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        } catch (IOException e) {
+            throw new ConversionException(e);
+        }
+
+        return result;
+    }
+
+    @Override
+    public TypedOutput toBody(Object o) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/repository/XmlConverter.java]: use try-with-resources

Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/repository/XmlConverter.java]: You can just throw `UnsupportedOperationException`

File: src/main/java/ru/yandex/qatools/allure/AllureConfig.java [hunk 3]
```diff
@@ -7,11 +8,13 @@
...
+import java.io.File;
+
 /**
  * eroshenkoam
  * 13/08/14
  */
-@Resource.File("${user.home}/.allure")
+@Resource.File("${user.home}/.allure/config")
 @With(MapOrSyspropPathReplacerProvider.class)
 public class AllureConfig {
```
Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/AllureConfig.java]: I don't like this config at all

**Comment by eroshenkoam** [comment]
test this please

**Comment by baev** [comment]
thanks!

**Pull request merged**